File: src/main/java/com/siat/web/notice/NoticeService.java
package com.siat.web.notice;



import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional 
public class NoticeService {

	@Autowired
	private NoticeRepository noticeRepository;
	
	public List<Notice> noticeList(){
		
		return noticeRepository.findAllDesc();
	}
	
	
	public void insertNotice(Notice notice) {
		noticeRepository.save(notice);
	}
	
	public Notice detailNotice(Notice notice) {
		return noticeRepository.findById(notice.getNotice_id()).get();
	}
	public void updateNotice(Notice notice) {
		Notice findNotice = noticeRepository.findById(notice.getNotice_id()).orElse(null);
	    if (findNotice != null) {
	        findNotice.setTitle(notice.getTitle());
	        findNotice.setContent(notice.getContent());

	        // 변경된 필드들을 저장
	        noticeRepository.save(findNotice);
	    }
	}
	public void deleteNotice(Notice notice) {
		noticeRepository.deleteById(notice.getNotice_id());
		
	}

	public void increaseCount(Long getNotice_id) {
		noticeRepository.increaseCount(getNotice_id);
	}

	public Notice getNoticeByNotice(Long getNotice_id) {
		return noticeRepository.findById(getNotice_id).orElse(null);
	}



}

File: react-siat-admin/src/service/NoticeService.js
import axios from "axios";

const NOTICE_API_BASE_URL = "http://localhost:8088/api/";

function noticeList(){
  return axios.get(NOTICE_API_BASE_URL + "noticeList");
}

function insertNotice(notice){
  return axios.post(NOTICE_API_BASE_URL + "insertNotice", notice);
}

function detailNotice(notice_id){
  return axios.get(NOTICE_API_BASE_URL + "detailNotice/" + notice_id)
}

function modifyNotice(notice_id, notice){
  return axios.post(NOTICE_API_BASE_URL + "modifyNotice/" + notice_id, notice)
}

function deleteNotice(notice_id){
  return axios.delete(NOTICE_API_BASE_URL + "noticeList/" + notice_id);
}




const NoticeService = {
  noticeList,
  insertNotice,
  detailNotice,
  modifyNotice,
  deleteNotice
};

export default NoticeService;

